### DISCUSSION TITLE
"Find Symbol" does not work for medium or large projects

### DISCUSSION THREAD

**Discussion by 12oclocker** [eranif/codelite#3556] [Setup and configuration issues]
### What happened?

IDE v 17.0.0, Using C++ and Linux Debian Stable, "Find Symbol" does not work for medium or large projects, nor does "Find References", it will work for some small projects and small files with very little lines of code, such as a 100 lines of code. If a project has large amounts of code, such as 3000 lines of code, it simply does not work. Restarting language server has no effect. I have went back to using GEANY as my GUI, because navigating a large project and not being able to jump to a declaration significantly increases development time. I have searched the internet and find similar complaints and a few closed bug reports saying this was fixed. But I have multiple projects and multiple computers all having this issue. I was not able to compile any new version of Code-lite on Debian stable. But I found no other reports of this bug being open.

### Version

17.0.0

### Operating system

Linux

### Steps to reproduce

```text
open any C++ project with around 3000 lines of code. "Find Symbol" and "Find References" does not work.
```

**Comment by eranif** [comment]
If it does not work, it means that your configuration is wrong. You need to configure the language server (which you forgot to mention which one is in use) in order for it to find symbols.

Follow the instructions here:

https://docs.codelite.org/plugins/lsp/#cmake-clangd-ctagsd
Upgrade your CodeLite to a more recent version (current version is 17.12) or build it from sources, hopefully we will release 18.0.0 very soon

P.S converting this issue to a discussion. As CodeLite works as good as the LSP it uses (which works perfectly for projects with millions of lines of code, including CodeLite itself)

**Comment by 12oclocker** [comment]
thanks for the quick response, in the language server settings, I have the defaults (shown below), I noticed "working directory" is blank, but it always had been, changing it to $(WorkspacePath)/ makes no difference
Compiler and Linker I set the libraries and compile options, but other than that I have not changed any defaults.
Using the default "Codelite Makefile Generator", Ive tried enabling only clangd or only ctagsd, and both together, and tried putting $(WorkspacePath) in the Working Directory: field. nothing seems to make a difference.
 was going to try 17.12 but could not find a bookworm repo on codelite website
Also I noticed ctagsd.json does Not include my source or header code files in search_path....
This is all it has...  "search_path": ["/usr/include", "/usr/include/c++/12", "/usr/include/c++/12/backward", "/usr/include/x86_64-linux-gnu", "/usr/include/x86_64-linux-gnu/c++/12", "/usr/lib/gcc/x86_64-linux-gnu/12/include", "/usr/local/include"],
Also I noticed... "compile_commands.json" is missing, not being generated when clangd is enabled. Even after workspace or project rebuild. And I have the option[X]"Generate compile_commands.json" checked in code completion dialog, but it does not generate the file...
So I also tried the cc_wrapper method... https://docs.codelite.org/misc/cc_wrapper/
And I get this error... ERROR: could not open file: /tmp/out.txt I could not find where Codelite Makefile Generator log file is located for setting export CC_COMPILATION_LOG
I also changed -limit-results=250 to --limit-results=250 and changed -header-insertion-decorators=0 to --header-insertion-decorators=0
but still having same issue...
Syntax Highlighting works, but Find Symbol does not.
I believe the issue with clangd is that compile_commands.json is not being generated, even though it is enabled in code completion dialog.
I dont know why ctagsd ctagsd.json search_path is missing my project paths, and adding $(WorkspacePath) in the Working Directory for laguage server options did not change anything in that regard.

clangd
[X] Enabled
Name: clangd
# Command to execute:
/usr/bin/clangd
  -limit-results=250
  -header-insertion-decorators=0
  --compile-commands-dir=$(WorkspacePath)
Working Directory:
Languages: c;cpp
Connection String: stdio
Priority: 90
[X] Display Diagnostics
__________________________________
ctagsd
[X] Enabled
Name: ctagsd
# Command to execute:
/usr/lib/codelite/ctagsd
  --port
  45634
  --log-level
  ERR
Working Directory:
Langauges: c;cpp
Connection String: tcp://127.0.0.1:45634
Priority: 50
[X] Display Diagnostics
